src/context/WalletProvider/MobileWallet/components/MobileCreate.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  Button,
+  Code,
+  FormControl,
+  FormErrorMessage,
+  Input,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Tag,
+  Wrap,
+} from '@chakra-ui/react'
+import * as native from '@shapeshiftoss/hdwallet-native'
+import { Revocable } from '@shapeshiftoss/hdwallet-native-vault/dist/util'
+import { range } from 'lodash'
+import { ReactNode, useEffect, useMemo, useRef, useState } from 'react'
+import { FaEye } from 'react-icons/fa'
+import { useTranslate } from 'react-polyglot'
+import { useHistory, useLocation } from 'react-router-dom'
+import { Text } from 'components/Text'
+
+import { mobileLogger } from '../config'
+import { addWallet, createWallet } from '../mobileMessageHandlers'
+import { RevocableWallet } from '../RevocableWallet'
+import { MobileLocationState } from '../types'
+
+const revocable = native.crypto.Isolation.Engines.Default.revocable
+
+const moduleLogger = mobileLogger.child({
+  namespace: ['WalletProvider', 'MobileWallet', 'components', 'MobileCreate'],
+})
+
+export const MobileCreate = () => {
+  const history = useHistory()
+  const location = useLocation<MobileLocationState>()
+  const [revealed, setRevealed] = useState<boolean>(false)
+  const revealedOnce = useRef<boolean>(false)
+  const handleShow = () => {
+    revealedOnce.current = true
+    setRevealed(!revealed)
+  }
+  const [words, setWords] = useState<ReactNode[] | null>(null)
+  const [vault, setVault] = useState<RevocableWallet | null>(null)
+  const [label, setLabel] = useState('')
+  const [isSaving, setIsSaving] = useState(false)
+  const [revoker] = useState(new (Revocable(class {}))())
+  const translate = useTranslate()
+
+  const placeholders = useMemo(() => {
+    return range(1, 13).map(i => (
+      <Tag
+        p={2}
+        flexBasis='30%'
+        justifyContent='flex-start'
+        fontSize='md'
+        colorScheme='blue'
+        key={i}
+      >
+        <Code mr={2}>{i}</Code>
+        •••••••
+      </Tag>
+    ))
+  }, [])
+
+  useEffect(() => {
+    try {
+      setVault(createWallet())
```

Comment:
possible benefit to separating these calls for error line numbers?

Reply:
`createWallet` shouldn't throw.